test/app/fiber.test.lua

```diff
@@ -255,6 +255,17 @@ fiber.self().storage.key -- our local storage is not affected by f
 -- attempt to access local storage of dead fiber raises error
 pcall(function(f) return f.storage end, f)
 
+--
+-- gh-6210: Access to fiber storage of current fiber via fiber.storage()
+--
+
+fiber.self().storage == fiber.storage()
```

Comment:
Thanks for your review. I'll sequentially answer all your questions.

First of all, this comment. No. This check gives a guarantee that fiber.storage() and fiber.self().storage return the same lua table. It's enough to understand (test) that changes in the first object reflect the second one and vice versa.
I could drop such checks or I could stay this test as is as "example" and visual demonstration that it works.

Following comments:
> As for me, saying "single way to do it was
fiber.self().storage" is not correct, because it sounds like "before that we could look at the attribute of the object only by looking at the attribute of the object".

I don't see anything bad. Could you propose your variant or change?

> Did I understand correctly that the main reason for adding functionality is syntactic sugar or the main reason is performance? If it was performance, do you have information on how much time the production code (in some cases) might be spending on these calls? I mean, how much can the performance of a real application grow, can it be> 1%?

I should describe how it works in one our project.
Fiber.storage stores "request context". Request context is uuid of request and some additional meta info. This request context is propagated over all netbox calls. The part of request context is info about user e.g. login/access role/... New version of our project introduces "multitenancy". So we add "tenant" value to request context. And here we faced with some issues.

Our code base use "vars" (there is open source implementation of vars in cartridge https://github.com/tarantool/cartridge/blob/master/cartridge/vars.lua). Vars is an abstraction under _G.
vars.value means "_G.__<module_name>.value_name" (maybe it's not completely exact but something similar). It's a cartridge and it works with single tenant in terms of my project.

To avoid overcomplication of my project I made vars multitenant. It means that currently "vars" is something like:
```
local request_context = fiber.self().storage.request_context.tenant
_G.__<module_name>.value_name[tenant].value_name
```

That mean that each value reference calls `fiber.self().storage`. It leads to some negative performance impact.
I have some "microbenchmarks" to test "hot" fragments of project code. This benchmarks show that some functions start to work 2-3 times slower if change construction like:
```lua
local value = vars.value
if value ~= nil then
...
end
```

to
```lua
local value = vars.value -- and use it in some place
if vars.value ~= then -- the second dereference
   ...
end
```

I think it's inappropriate for similar similar case. I assume it could give up to ~5% performance loss in overall result and also increases GC pressure. It's my motivation to introduce such function. But yes, it could be considered as syntactic sugar.

Reply:
> Thanks for your review. I'll sequentially answer all your questions.

Hi)

> 
> First of all, this comment. No. This check gives a guarantee that fiber.storage() and fiber.self().storage return the same lua table. It's enough to understand (test) that changes in the first object reflect the second one and vice versa.
> I could drop such checks or I could stay this test as is as "example" and visual demonstration that it works.

Up to you.  I think any option is normal

> 
> Following comments:
> 
> > As for me, saying "single way to do it was
> > fiber.self().storage" is not correct, because it sounds like "before that we could look at the attribute of the object only by looking at the attribute of the object".
> 
> I don't see anything bad. Could you propose your variant or change?
> 

Maybe something like: "in addition to `fiber.self().storage` an alternative way to access the storage of the current fiber has been added ..."
Up to you.

> > Did I understand correctly that the main reason for adding functionality is syntactic sugar or the main reason is performance? If it was performance, do you have information on how much time the production code (in some cases) might be spending on these calls? I mean, how much can the performance of a real application grow, can it be> 1%?
> 
> I should describe how it works in one our project.
> Fiber.storage stores "request context". Request context is uuid of request and some additional meta info. This request context is propagated over all netbox calls. The part of request context is info about user e.g. login/access role/... New version of our project introduces "multitenancy". So we add "tenant" value to request context. And here we faced with some issues.
> 
> Our code base use "vars" (there is open source implementation of vars in cartridge https://github.com/tarantool/cartridge/blob/master/cartridge/vars.lua). Vars is an abstraction under _G.
> vars.value means "_G.__<module_name>.value_name" (maybe it's not completely exact but something similar). It's a cartridge and it works with single tenant in terms of my project.
> 
> To avoid overcomplication of my project I made vars multitenant. It means that currently "vars" is something like:
> 
> ```
> local request_context = fiber.self().storage.request_context.tenant
> _G.__<module_name>.value_name[tenant].value_name
> ```
> 
> That mean that each value reference calls `fiber.self().storage`. It leads to some negative performance impact.
> I have some "microbenchmarks" to test "hot" fragments of project code. This benchmarks show that some functions start to work 2-3 times slower if change construction like:
> 
> ```lua
> local value = vars.value
> if value ~= nil then
> ...
> end
> ```
> 
> to
> 
> ```lua
> local value = vars.value -- and use it in some place
> if vars.value ~= then -- the second dereference
>    ...
> end
> ```
> 
> I think it's inappropriate for similar similar case. I assume it could give up to ~5% performance loss in overall result and also increases GC pressure. It's my motivation to introduce such function. But yes, it could be considered as syntactic sugar.

As I write before - LGTM.

P.S.
I am OK with such extending of API, but AFAIU @Mons has a slightly different opinion. I think you should talk to him or @unera